tests/Unit/ParallelAlgorithms/SurfaceFinder/Test_SurfaceFinder.cpp

```diff
@@ -0,0 +1,105 @@
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Creators/Sphere.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/InterfaceLogicalCoordinates.hpp"
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "NumericalAlgorithms/Spectral/LogicalCoordinates.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/YlmSpherepack.hpp"
+#include "ParallelAlgorithms/SurfaceFinder/SurfaceFinder.hpp"
+#include "PointwiseFunctions/MathFunctions/Gaussian.hpp"
+
+SPECTRE_TEST_CASE("Unit.ParallelAlgorithms.SurfaceFinder.SurfaceFinder",
+                  "[Unit][ParallelAlgorithms]") {
+  MathFunctions::Gaussian<3, Frame::Inertial> gaus{
+      10., 5., std::array<double, 3>{0., 0., -1.}};
+  const double target = 4.;
+  const double target_bulge = 2.5;
+
+  size_t ang_ref = 4;
+  const size_t dim = 3;
+  const std::array<size_t, 2> grid_points_r_angular{{10, ang_ref}};
+  const double inner_radius = 1.;
+  const double outer_radius = 5.;
+  const double sphericity = 0.5;
+  const size_t ref = 0;
+  const domain::creators::Sphere sphere_helper{
+      inner_radius, outer_radius, sphericity, ref, grid_points_r_angular, true};
+  const std::vector<std::array<size_t, dim>> ref_levels{
+      sphere_helper.initial_refinement_levels()};
+  const auto extents = sphere_helper.initial_extents();
+  Domain<dim> dom{sphere_helper.create_domain()};
+
+  std::vector<ElementId<dim>> ids = initial_element_ids(ref_levels);
+  const size_t N_elems = ids.size();
+  const auto quadrature = Spectral::Quadrature::GaussLobatto;
+
+  std::unordered_map<ElementId<dim>, ElementMap<dim, Frame::Inertial>>
+      element_map_map(N_elems);
+  std::unordered_map<ElementId<dim>, Element<dim>> elements_map(N_elems);
+  std::unordered_map<ElementId<dim>, Mesh<dim>> mesh_map;
+  std::unordered_map<ElementId<dim>, Scalar<DataVector>> radii_map;
+  std::unordered_map<ElementId<dim>, Scalar<DataVector>> data_map;
+  std::unordered_map<ElementId<dim>, Index<3>> extents_map;
+  std::unordered_map<ElementId<dim>,
+                     tnsr::I<DataVector, dim, Frame::ElementLogical>>
+      logical_coords_map(N_elems);
+  std::unordered_map<ElementId<dim>, tnsr::I<DataVector, dim, Frame::Inertial>>
+      inertial_coords_map(N_elems);
+
+  for (size_t i = 0; i < N_elems; i++) {
```

Comment:
Either set up just a single element and find a contour on it, or set up a full sphere and construct a Strahlkorper from the full surface. Or both :) I think both would actually be very nice: single element as a unit test, and constructing a Strahlkorper as an integration test because that's how we want to use this code, so we want to make sure everything fits together.